### DISCUSSION TITLE
esp8266 code,help me!

### DISCUSSION THREAD

**Discussion by Roycalonge** [revoxhere/duino-coin#1206] [General]
Arduino:1.8.19 (Windows Store 1.8.57.0) (Windows 10), Tarjeta:"Generic ESP8266 Module, 80 MHz, Flash, Disabled (new aborts on oom), Disabled, All SSL ciphers (most compatible), 32KB cache + 32KB IRAM (balanced), Use pgm_read macros for IRAM/PROGMEM, dtr (aka nodemcu), 26 MHz, 40MHz, DOUT (compatible), 4MB (FS:3MB OTA:~512KB), 2, nonos-sdk 2.2.1+100 (190703), v2 Lower Memory, Disabled, None, Only Sketch, 115200"

C:\ESP8266_Code\ESP8266_Code.ino: In function 'String {anonymous}::httpGetString(String)':

ESP8266_Code:77:10: error: 'class WiFiClient' has no member named 'setInsecure'

   77 |   client.setInsecure();

      |          ^~~~~~~~~~~

C:\ESP8266_Code\ESP8266_Code.ino: In function 'void {anonymous}::SetupWifi()':

ESP8266_Code:148:8: error: 'class WiFiClass' has no member named 'mode'

  148 |   WiFi.mode(WIFI_STA); // Setup ESP in client mode

      |        ^~~~

ESP8266_Code:148:13: error: 'WIFI_STA' was not declared in this scope

  148 |   WiFi.mode(WIFI_STA); // Setup ESP in client mode

      |             ^~~~~~~~

ESP8266_Code:149:8: error: 'class WiFiClass' has no member named 'setSleepMode'

  149 |   WiFi.setSleepMode(WIFI_NONE_SLEEP);

      |        ^~~~~~~~~~~~

ESP8266_Code:149:21: error: 'WIFI_NONE_SLEEP' was not declared in this scope

  149 |   WiFi.setSleepMode(WIFI_NONE_SLEEP);

      |                     ^~~~~~~~~~~~~~~

ESP8266_Code:150:14: error: invalid conversion from 'const char*' to 'char*' [-fpermissive]

  150 |   WiFi.begin(SSID, PASSWORD);

      |              ^~~~

      |              |

      |              const char*

In file included from C:\ESP8266_Code\ESP8266_Code.ino:1:

C:\Program Files\WindowsApps\ArduinoLLC.ArduinoIDE_1.8.57.0_x86__mdqgnx93n4wtt\libraries\WiFi\src/WiFi.h:79:21: note:   initializing argument 1 of 'int WiFiClass::begin(char*, const char*)'

   79 |     int begin(char* ssid, const char *passphrase);

      |               ~~~~~~^~~~

ESP8266_Code:153:15: error: 'class WiFiClass' has no member named 'waitForConnectResult'

  153 |   while (WiFi.waitForConnectResult() != WL_CONNECTED) {

      |               ^~~~~~~~~~~~~~~~~~~~

ESP8266_Code:157:18: error: invalid conversion from 'const char*' to 'char*' [-fpermissive]

  157 |       WiFi.begin(SSID, PASSWORD);

      |                  ^~~~

      |                  |

      |                  const char*

In file included from C:\ESP8266_Code\ESP8266_Code.ino:1:

C:\Program Files\WindowsApps\ArduinoLLC.ArduinoIDE_1.8.57.0_x86__mdqgnx93n4wtt\libraries\WiFi\src/WiFi.h:79:21: note:   initializing argument 1 of 'int WiFiClass::begin(char*, const char*)'

   79 |     int begin(char* ssid, const char *passphrase);

      |               ~~~~~~^~~~

C:\ESP8266_Code\ESP8266_Code.ino: In function 'void {anonymous}::VerifyWifi()':

ESP8266_Code:224:10: error: 'class WiFiClass' has no member named 'reconnect'; did you mean 'disconnect'?

  224 |     WiFi.reconnect();

      |          ^~~~~~~~~

      |          disconnect

C:\ESP8266_Code\ESP8266_Code.ino: In function 'void {anonymous}::ConnectToServer()':

ESP8266_Code:270:36: error: no matching function for call to 'WiFiClient::connect(String&, int&)'

  270 |   while (!client.connect(host, port));

      |                                    ^

In file included from C:\Program Files\WindowsApps\ArduinoLLC.ArduinoIDE_1.8.57.0_x86__mdqgnx93n4wtt\libraries\WiFi\src/WiFi.h:31,

                 from C:\ESP8266_Code\ESP8266_Code.ino:1:

C:\Program Files\WindowsApps\ArduinoLLC.ArduinoIDE_1.8.57.0_x86__mdqgnx93n4wtt\libraries\WiFi\src/WiFiClient.h:34:15: note: candidate: 'virtual int WiFiClient::connect(IPAddress, uint16_t)'

   34 |   virtual int connect(IPAddress ip, uint16_t port);

      |               ^~~~~~~

C:\Program Files\WindowsApps\ArduinoLLC.ArduinoIDE_1.8.57.0_x86__mdqgnx93n4wtt\libraries\WiFi\src/WiFiClient.h:34:33: note:   no known conversion for argument 1 from 'String' to 'IPAddress'

   34 |   virtual int connect(IPAddress ip, uint16_t port);

      |                       ~~~~~~~~~~^~

C:\Program Files\WindowsApps\ArduinoLLC.ArduinoIDE_1.8.57.0_x86__mdqgnx93n4wtt\libraries\WiFi\src/WiFiClient.h:35:15: note: candidate: 'virtual int WiFiClient::connect(const char*, uint16_t)'

   35 |   virtual int connect(const char *host, uint16_t port);

      |               ^~~~~~~

C:\Program Files\WindowsApps\ArduinoLLC.ArduinoIDE_1.8.57.0_x86__mdqgnx93n4wtt\libraries\WiFi\src/WiFiClient.h:35:35: note:   no known conversion for argument 1 from 'String' to 'const char*'

   35 |   virtual int connect(const char *host, uint16_t port);

      |                       ~~~~~~~~~~~~^~~~

exit status 1

'class WiFiClient' has no member named 'setInsecure'

Este informe podría contener más información con
"Mostrar salida detallada durante la compilación"
opción habilitada en Archivo -> Preferencias.

**Comment by BBS215** [comment]
1) Try to update your ESP8266 libraries:
Open Boards Manager from Tools> Board menu and find esp8266 platform.
Select the version 3.0.2 from a drop-down box.
Click install or update button.
2) Try to use arduino IDE not from the windows store, but from the official arduino website: https://www.arduino.cc/en/software

**Reply by Roycalonge** [reply]
Hello, I have updated the libraries, the ide was installed from the official arduino page, the error is thrown at me just by clicking to verify

**Reply by BBS215** [reply]
I apologize, I expressed myself incorrectly. Install arduino ide from here: https://downloads.arduino.cc/arduino-1.8.19-windows.exe
The windows store version may contain bugs.

**Reply by Roycalonge** [reply]
Yes, thanks bro, I already solved it. I only had to download the source code and the ide of the download options, thanks for your contribution, the truth was I was thinking that my card did not work

**Comment by Roycalonge** [comment]
SPIFFSEditor.h: No such file or directory
help pls!

**Reply by BBS215** [reply]
This is strange ... ESP8266_Code.ino does not use the ESPAsyncWebServer library, which has the SPIFFSEditor.h file. You can try to take the file from here, but I doubt it will help: https://github.com/me-no-dev/ESPAsyncWebServer/tree/master/src Please post the complete project build log.

**Reply by Roycalonge** [reply]
hi bro, again thanks for your help, what I'm trying to do is set esp as host, download the code from this link https://github.com/ricaun/DuinoCoinI2C and try to follow the steps of a video tutorial of yt https://www.youtube.com/watch?v=swvrdkY02uc&t=43s but when verifying it throws an error:
Arduino: 1.8.20 Hourly Build 2021/12/20 07:33 (Windows 10), Card: "NodeMCU 1.0 (ESP-12E Module), 80 MHz, Flash, Disabled (new aborts on oom), Disabled, All SSL ciphers (most compatible), 32KB cache + 32KB IRAM (balanced), Use pgm_read macros for IRAM / PROGMEM, 4MB (FS: 2MB OTA: ~ 1019KB), 2, v2 Lower Memory, Disabled, None, Only Sketch, 115200 "

DoinoCoin_AsyncWebServer: 20:10: fatal error: SPIFFSEditor.h: No such file or directory

   20 | #include <SPIFFSEditor.h>

      | ^ ~~~~~~~~~~~~~~~

compilation terminated.

exit status 1

SPIFFSEditor.h: No such file or directory



This report could contain more information with
"Show verbose output during compilation"
option enabled in File -> Preferences.

**Reply by Roycalonge** [reply]
ready bro, thank you very much for your valuable help

**Reply by BBS215** [reply]
It's clearer that way.
1) Go to https://github.com/me-no-dev/ESPAsyncWebServer
Click Code - Download ZIP.
In Arduino IDE click Sketch - Include Library - Add .ZIP Library - select the downloaded file.

2) Do the same for the following libraries:
https://github.com/me-no-dev/ESPAsyncTCP
https://github.com/bblanchon/ArduinoJson
https://github.com/ThingPulse/esp8266-oled-ssd1306/

3) Make sure the correct board is selected.

4) Click Verify.

5) If there are errors, add the library in the same way:
https://github.com/esp8266/Arduino

**Reply by Roycalonge** [reply]
![Captura de pantalla 2022-01-06 232119](https://user-images.githubusercontent.com/71743157/148491236-6d9c477e-39dd-4a77-b8a3-1ce845189d2a.png)
I already downloaded and included all the libraries and I have fixed all the errors that it threw me but now after compiling I go to the monitor and it throws me this error that I do not know why:
Connected to WiFi!
Local IP address: 192.168.0.18
[HTTP] GET ... failed, error: connection failed
[] FreeRam: xxxx I2C []

**Reply by BBS215** [reply]
File DuinoCoin_Pool.ino
Replace string 17:
const char * urlPool = "http://51.15.127.80:4242/getPool";
to
const char * urlPool = "http://51.15.127.80/getPool";

**Reply by Roycalonge** [reply]
ready bro, thank you very much for your valuable help

**Reply by Roycalonge** [reply]
hi bro, the server is down? I got the error again!

**Reply by BBS215** [reply]
Server is down now, I got the same error.

**Reply by alsurya** [reply]
Hello guys, I got the same problem with error in this below

 error: 'class WiFiClass' has no member named 'setSleepMode'

149 | WiFi.setSleepMode(WIFI_NONE_SLEEP);

Please, Anyone can help me for solving this error and any libraries I must install?
